pkg/token/serviceclient/handler.go

```diff
@@ -20,7 +20,7 @@ var _ common.TokenChannelInterface = (*Handler)(nil)
 
 //go:generate mockgen -build_flags=-mod=mod -destination=../../mocks/IdentityAPI_mocks.go -package=mocks github.com/hewlettpackard/hpegl-provider-lib/pkg/token/serviceclient IdentityAPI
 type IdentityAPI interface {
-	GenerateToken(context.Context, string, string, string) (string, error)
+	GenerateToken(context.Context, string, string, string, string) (string, error)
```

Comment:
Would it be possible for the final parameter here to be `IAMVersion`?

Reply:
It is, but I'll fix up the follow-up function called to put `IAMVersion` as the last parameter.